src/Soil/rre.jl

```diff
@@ -92,14 +92,29 @@ function ClimaLSM.make_rhs(model::RichardsModel)
     function rhs!(dY, Y, p, t)
         @unpack ν, vg_α, vg_n, vg_m, K_sat, S_s, θ_r = model.parameters
         top_flux_bc, bot_flux_bc =
-            boundary_fluxes(model.boundary_conditions, p, t)
+            boundary_fluxes(model.boundary_conditions, p, t) # these are floats
         z = model.coordinates.z
         interpc2f = Operators.InterpolateC2F()
         gradc2f_water = Operators.GradientC2F()
+
+        # We are setting a boundary value on a flux, which is a gradient of a scalar
+        # Therefore, we should set boundary conditions in terms of a covariant vector
+        # We set the third component first - supply a Covariant3Vector
+
+        # Without topography only
+        # In Cartesian coordinates, W (z^) = Cov3 (z^)= Contra3 (n^ = z^)
+        # In spherical coordinates, W (r^) = Cov3 (r^) = Contra3 (n^ = r^)
+
+        # It appears that the WVector is converted internally to a Covariant3Vector for the gradient value
+        # at the boundary. Offline tests indicate that you get the same thing if
+        # the bc is WVector(F) or Covariant3Vector(F*Δr) or Contravariant3Vector(F/Δr)
```

Comment:
Feel free to keep or remove these, whichever you feel is more useful at this stage.